lib/ui/login/login_view_model.dart

```diff
@@ -0,0 +1,37 @@
+import 'package:flutter_survey/model/login_model.dart';
+import 'package:flutter_survey/usecases/base/base_use_case.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:flutter_survey/ui/login/login_state.dart';
+import 'package:flutter_survey/usecases/store_auth_token_use_case.dart';
+import 'package:flutter_survey/usecases/log_in_use_case.dart';
+
+class LoginViewModel extends StateNotifier<LoginState> {
+  final LogInUseCase _logInUseCase;
+  final StoreAuthTokenUseCase _storeAuthTokenUseCase;
+
+  LoginViewModel(
+    this._logInUseCase,
+    this._storeAuthTokenUseCase,
+  ) : super(const LoginState.init());
+
+  void logIn(String email, String password) async {
+    state = const LoginState.loading();
+    final LoginInput input = LoginInput(
+      email: email,
+      password: password,
+    );
+    final result = await _logInUseCase.call(input);
+    if (result is Success<LoginModel>) {
+      final LoginModel loginData = result.value;
+      try {
+        _storeAuthTokenUseCase.save(loginData);
+        state = const LoginState.loginSuccess();
+      } catch (exception) {
+        state = LoginState.loginError((exception as Failed).getErrorMessage());
+      }
```

Comment:
Then, we should deal with this `storeAuthTokenUseCase` same as `logInUseCase` instead of wrapping it inside a try/catch block